.github/workflows/main.yml

```diff
@@ -13,23 +13,53 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
 
     name: Compile and Run Tests
     steps:
       - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
```

Comment:
Since the version of it that is being downloaded and built here already has a meson.build added to it, the best way to fukfill this request would be to add it as a meson wrap and then meson will download it if needed without any user interaction.

Reply:
Also I'm pretty sure the owner of the repo that wolfv/agp-standalone is forked from, would be happy to accept a meson PR. :D

Reply:
Yeah, having taken a second look at the project (which I originally assumed was much smaller), I think a meson wrap would be ideal.  Also, @wolfv, if you were willing to at least try to get your changes merged back upstream, it would reduce the number of `argp-standalone` forks and (hopefully) mean any security fixes would automatically come to us.

Reply:
yeah, there are a number of argp-standalone forks, indeed. None that I could find had Windows support and most are using ancient automake scripts ... I wouldn't really count on any fixes :)

I think there are probably some cross-platform C CLI parsing libraries available -- and there are definitely some kinda nice C++ ones (header only, even) such as [CLI11](https://github.com/CLIUtils/CLI11) that we are using extensively.

My plan was to package argp-standalone for Windows as well in the conda-forge repository, and then we can install it the same way as on macOS using `mamba` (from the `environment.yml` file).

Reply:
I really don't want to make `zchunk` dependent on `mamba` to build and test.  I need to take another look at the macOS build, but I would far prefer that we use something like meson wrap for this kind of situation so we're not increasing the number of dependencies.

Reply:
`mamba` is just a package manager like `brew`, `apt`, `dnf` ... where do you get `meson`, `libcurl` and `zstd-devel` from on your Jenkins build? If you want more details on how `mamba` and `conda-forge` work, I'd be happy to give you more insights.

Reply:
No, I understand where you're coming from, and, on second thought, I think using `mamba` in the GitHub actions is fine.  What I don't want is a hard requirement that anyone building zchunk must use `mamba` or `dnf` or `apt` or any other package manager.  As for Jenkins, my policy is to use the default package manager for the system, but that doesn't really apply for Windows.

Going back to `argp-standalone`, I'm reminded a bit of [the XKCD comic about standards](https://xkcd.com/927). :grin: If you really want to maintain a new fork, that's fine, but I think it might be worth trying to get your changes back upstream.

Reply:
Totally agree. The meson file in zchunk shouldn't ever require a specific package manager!
Yeah, the argp situation is really unfortunate. It really doesn't seem to be properly supported in a cross-platform fashion.
I am also not "happy" to have to maintain my own fork -- I did open a PR to ericonr's fork: https://github.com/ericonr/argp-standalone/pull/1

The problem is that it's just one of a couple `argp-standalone` "forks" on github and there is no clear, dominant "winner" :)

Reply:
Ok, sounds good.  If ericonr accepts your PR, we can switch upstream on this.  And, if you can get argp-standalone into `mamba`, we'll just switch to using it there.

Reply:
argp-standalone is needed by any linux distros and non-linux POSIX operating systems that want to package software which depends on argp but doesn't use the GNU libc. Particularly that fork was made by a maintainer of a musl libc distro.

According to https://repology.org/project/argp-standalone/versions I believe there are, effectively, exactly two versions of it around. One is the original unmaintained one, and one is the fork you found.

The reason I suggested you submit a PR to it is because I hang out in the same place as the author on IRC, know that the author likes meson, and also... I mentioned it and got an explicit response of interest, albeit:

> Why do people not split commits :((
> You don't fix whitespace and utf-8 in the same commit you add a new build system